src/ser/serde.rs

```diff
@@ -258,21 +303,40 @@ impl ser::Serializer for Serializer {
     where
         T: Serialize,
     {
-        if name == UUID_NEWTYPE_NAME {
-            match value.serialize(self)? {
-                Bson::String(s) => {
-                    // the serializer reports itself as human readable, so `Uuid` will
-                    // serialize itself as a string.
-                    let uuid = crate::Uuid::parse_str(s).map_err(Error::custom)?;
-                    Ok(Bson::Binary(uuid.into()))
+        match name {
+            UUID_NEWTYPE_NAME => {
+                match value.serialize(self)? {
+                    Bson::String(s) => {
+                        // the serializer reports itself as human readable, so `Uuid` will
```

Comment:
This is no longer an invariant, no?

Reply:
good catch, forgot about this. Fixed and added a test case for this.